backend/module/chaos/serverexperimentation/xds/xds_cache.go

```diff
@@ -77,36 +82,59 @@ func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCa
 
 	// Create/Update experiments
 	for cluster, experiments := range clusterFaultMap {
+		resources := make(map[gcpTypes.ResponseType][]gcpTypes.ResourceWithTtl)
+
 		logger.Debugw("Injecting fault for cluster", "cluster", cluster)
 
-		runtimeResource := generateRTDSResource(experiments, rtdsConfig, ttl, logger)
-		err := setSnapshot(runtimeResource, cluster, snapshotCache, logger)
+		// Enable ECDS if cluster present in the map
+		if _, exists := ecdsConfig.enabledClusters[cluster]; exists {
+			resources[gcpTypes.ExtensionConfig] = generateECDSResource(experiments, ttl, logger)
+
+			// generate empty Runtime config to clear any existing RTDS faults
+			resources[gcpTypes.Runtime] = generateRTDSResource([]*experimentation.Experiment{}, rtdsConfig, ttl, logger)
+		} else {
+			resources[gcpTypes.Runtime] = generateRTDSResource(experiments, rtdsConfig, ttl, logger)
+			resources[gcpTypes.ExtensionConfig] = generateEmptyECDSResource(cluster, ecdsConfig, logger)
+		}
+
+		err := setSnapshot(resources, cluster, snapshotCache, logger)
 		if err != nil {
 			logger.Errorw("Unable to set the fault for cluster", "cluster", cluster,
 				"error", err)
 		}
 	}
 }
 
-func setSnapshot(resource []gcpTypes.ResourceWithTtl, cluster string, snapshotCache gcpCacheV3.SnapshotCache, logger *zap.SugaredLogger) error {
-	computedVersion, err := computeChecksum(resource)
-	if err != nil {
-		return err
-	}
-
+func setSnapshot(resourceMap map[gcpTypes.ResponseType][]gcpTypes.ResourceWithTtl, cluster string, snapshotCache gcpCacheV3.SnapshotCache, logger *zap.SugaredLogger) error {
 	currentSnapshotVersion := ""
 	currentSnapshot, err := snapshotCache.GetSnapshot(cluster)
 	if err == nil {
-		currentSnapshotVersion = currentSnapshot.GetVersion(cluster)
+		currentSnapshotVersion, _ = computeChecksum(currentSnapshot)
```

Comment:
Seems unnecessary to recompute the checksum here? Would it be cleaner to compute the checksum of each resource type and set the snapshot if any of them are different?